montrek-software/montrek
# 277 reportingframework

## Background

- A new class `MontreReportManager` has been introduced and as a daughter class `LatexReportManager`, which opens the possibility to see pdf reports as part of the reporting, but not necessarilly all.
- The Latex Report is attached to any `MontrekListView` , such that every table can be generated as pdf at all times.
- The github workflow yml had to be extended such that pdf can be generated during testing
## Requirements

- XeLatex as Latex compiler has to be installed. In case of error do 

```
sudo apt-get install texlive-xetex texlive-fonts-recommended
```

- For Linux systems the Arial fonts are not installed necessarily, since these are windows fonts. If during pdf generation an error occurs, that Arial was not found, do

```
        sudo apt-get install -y ttf-mscorefonts-installer
        sudo fc-cache -f -v
```

## How to test

Go to any list view and click on the pdf generation button:

![image](https://github.com/chrishombach/montrek/assets/29869670/2698db95-8d3e-44bd-8b5a-4c58777e5574)

The table should be shown in a pdf:

![image](https://github.com/chrishombach/montrek/assets/29869670/c997b796-5c4b-4ebb-8c48-548f2ee77558)

## To Dos

- Latex is not yet installed in docker
- The github workflows can be made more efficient if the installation process is cached
- The currency list can not be compiled, since all the currency symbols have to be converted to latex code
